core/services/feeds/service.go

```diff
@@ -1550,6 +1553,32 @@ func (s *service) isRevokable(propStatus JobProposalStatus, specStatus SpecStatu
 	return propStatus != JobProposalStatusDeleted && (specStatus == SpecStatusPending || specStatus == SpecStatusCancelled)
 }
 
+func (s *service) periodicallySyncNodeInfo(ctx context.Context) {
```

Comment:
I think there is also an assumption in code where JD is connected to the core node when `SyncNodeInfo` is called, else it will return the error `could not fetch client` which may not be a big deal, just noise. But if we could check if the nodes are connected before synching, that would be nice.